pkg/server/api.go

```diff
@@ -340,3 +333,62 @@ func (h *defaultHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusMethodNotAllowed)
 	return
 }
+
+// getHTTPServerCfg returns the basic HTTP Server
+func getHTTPServerCfg(addr string, handler http.Handler) *http.Server {
+	return &http.Server{
+		Addr:    addr,
+		Handler: handler,
+		// CVE-2016-2183: disable http/2, which by definition, requires insecure ciphers
+		// Per https://golang.org/src/net/http/doc.go this is the runtime method of disabling HTTP/2
+		TLSNextProto: make(map[string]func(*http.Server, *tls.Conn, http.Handler)),
+		// We don't want to allow 1.1 as that's old.  This was flagged in a security audit.
+		TLSConfig: &tls.Config{
+			MinVersion:   tls.VersionTLS12,
+			CipherSuites: cipherOrder(),
+		},
+	}
+
+}
+
+// Disable insecure cipher suites for CVE-2016-2183
+// cipherOrder returns an ordered list of Ciphers that are considered secure
+// Deprecated ciphers are not returned.
+func cipherOrder() []uint16 {
+	var first []uint16
+	var second []uint16
+
+	allowable := func(c *tls.CipherSuite) bool {
+		// Disallow block ciphers using straight SHA1
+		// See: https://tools.ietf.org/html/rfc7540#appendix-A
+		if strings.HasSuffix(c.Name, "CBC_SHA") {
```

Comment:
Since we are matching with Suffix here, it won't match ciphers listed in blacklist that ends with `CBC_SHA256`, `CBC_SHA384` . Are these safe?

Reply:
SHA256, 384, and 512 are considered secure....for now.